app/models/mixins/authentication_mixin.rb

```diff
@@ -12,6 +12,33 @@ def self.authentication_check_schedule
       assocs = zone.respond_to?(assoc) ? zone.send(assoc) : []
       assocs.each { |a| a.authentication_check_types_queue(:attempt => 1) }
     end
+
+    def self.queue_authentication_check(args, user_id, zone)
```

Comment:
@jntullo I agree, the UI needs to know these results before proceeding as we do not allow the object to be saved until the credentials are validated.  We would need to change the code to "spin" the UI while waiting on the asynchronous task.  Unless @AparnaKarve thinks this can be done in the short term, I say we block for now and address this in a future update.